jenkins/vulnerability-scan/whitesource-report.jenkinsfile

```diff
@@ -0,0 +1,49 @@
+pipeline {
+    agent none
+    options {
+        timeout(time: 1200, unit: 'SECONDS')
+    }
+    stages {
+        stage('Retrieve vulnerability report') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+                    image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211028'
+                    args '-e LANG=en_US.UTF-8'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    withCredentials([
+                        string(credentialsId: 'whitesource-scan-bot-userkey', variable: 'wss_bot_userkey'),
+                        string(credentialsId: 'whitesource-scan-opensearch-opensearch-projectkey', variable: 'wss_os_os_projectkey')
+                    ]) {
+                        wss_api_call()
+                    }
+                }
+            }
+            post() {
+                always {
+                    cleanWs disableDeferredWipeout: true, deleteDirs: true
+                }
+            }
+        }
+    }
+}
+
+void wss_api_call() {
+    def toJson = {
+        input ->
+        groovy.json.JsonOutput.toJson(input)
+    }
+    def payloadBody = [
+        requestType: "getProjectVulnerabilityReport",
```

Comment:
We might need to use getProductInventoryReport API rather than getProjectVulnerabilityReport to get the entire dependencies.

Reply:
@ryanbogan please confirm.
Thanks.

Reply:
Ref to these API call [here](https://whitesource.atlassian.net/wiki/spaces/WD/pages/786137094/HTTP+API+v1.2).
I'm not sure if the `getProjectInventoryReport` contains vulnerability information; or we could possibly use both calls: one for dependencies tree, and the other for vulnerability report?

Reply:
If that has the both the vulnerabilities and the dependencies, then we should use getProjectInventory Report instead

Reply:
Move the httprequests to its own lib and add flexibility for more requestType, with tests.